ddl/db_test.go

```diff
@@ -164,6 +164,31 @@ func (s *testDBSuite4) TestAddIndexWithPK(c *C) {
 	s.tk.MustQuery("select * from test_add_index_with_pk2").Check(testkit.Rows("1 1 1 1", "2 2 2 2"))
 }
 
+func (s *testDBSuite5) TestAddIndexWithDupIndex(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use " + s.schemaName)
+
+	err1 := ddl.ErrDupKeyName.GenWithStack("index already exist %s; "+
+		"a background job is trying to add the same index, "+
+		"please check by `ADMIN SHOW DDL JOBS`", "idx")
+	err2 := ddl.ErrDupKeyName.GenWithStack("index already exist %s", "idx")
+
+	// When there is already an duplicate index, show error message.
+	tk.MustExec("create table test_add_index_with_dup (a int, key idx (a))")
+	_, err := tk.Exec("alter table test_add_index_with_dup add index idx (a)")
+	c.Check(errors.Cause(err1).(*terror.Error).Equal(err), Equals, true)
```

Comment:
Why here the error is `err1`?

Reply:
@wangggong 
It seems that the `idx` is public state. So I think here the error should be `err2`.

Reply:
You're right, the error check is a careless mistake.

The check can still pass because `err1` and `err2` are both `ddl.ErrDupKeyName`, the error message is actually ignored when comparing.

Reply:
Yes, thx a lot. I'll fix the method to check the error message. Sorry for the mistake...